pkgs/development/libraries/qt-6/functions/broken-mkspecs-files-restore.sh

```diff
@@ -0,0 +1,65 @@
+# https://github.com/NixOS/nixpkgs/issues/214765
+
+# the functions qtPrepareTool and qtPrepareLibExecTool are defined in
+# qtbase-dev/mkspecs/features/qt_functions.prf
+
+brokenMkspecsFilesRestore() {
+  # restore disabled mkspecs files to this derivation
+  # from brokenMkspecsFilesDisable in qtbase.
+  local source_prefix=$1 # ${qtbase.dev}
+  local target_prefix=$2 # $dev
+  local instloc_expected=$3 # $dev/bin or $dev/libexec
+  local funcname_expected=$4 # qtPrepareTool or qtPrepareLibExecTool
+  local mkspecs_path=mkspecs/features
+  local mkspecs_file_pattern='*.prf.bak' # note: .bak
+  local debug=${DEBUG:-false}
+  local dryrun=${DRYRUN:-false}
+  local dryrun_cmd=
+  if $dryrun; then dryrun_cmd=echo; fi
+  while read -r file_match_line; do
+    # note: mkspecs_file_bak != mkspecs_file
+    IFS=, read -r mkspecs_file_bak funcname variable \
+      default_tool suffix prepare instloc \
+      < <(sed -E -e 's/\)$//' -e 's/, /,/g' -e 's/[:(]/,/g' <<<"$file_match_line")
```

Comment:
Is it possible to de-duplicate some of these logic? We can keep them in the same .sh file. (Ideally it should be under the hooks directory, why functions)

Reply:
> Is it possible to de-duplicate some of these logic?

that was my plan ; )

hope its more readable now